tools/topology/topology2/cavs-rt5682-rt1019.conf

```diff
@@ -0,0 +1,283 @@
+<include/common/vendor-token.conf>
+<include/common/tokens.conf>
+<include/common/manifest.conf>
+<include/common/pcm.conf>
+<include/common/pcm_caps.conf>
+<include/common/fe_dai.conf>
+<include/common/route.conf>
+<include/components/host.conf>
+<include/components/dai.conf>
+<include/components/pipeline.conf>
+<include/components/copier.conf>
+<include/dais/ssp.conf>
+<include/dais/dmic.conf>
+<include/dais/pdm_config.conf>
+<include/common/data.conf>
+<include/dais/hw_config.conf>
+<include/pipelines/cavs/passthrough-capture.conf>
+<include/pipelines/volume-playback.conf>
+<include/pipelines/volume-capture.conf>
+
+#
+# List of all DAIs
+#
+# PCM0 ----> volume -----> SSP1  (Speaker - ALC1019)
+# PCM1 <---> volume <----> SSP0  (Headset - ALC5682)
+# PCM3 <---- volume <---- DMIC01 (dmic 48k capture)
+Object.Dai {
+	SSP."0" {
+		id 		0
+		direction	"duplex"
+		name		SSP0-Codec
+		default_hw_conf_id	0
+		sample_bits		16
+
+		Object.Base.hw_config."SSP0" {
+			id	0
+			mclk_freq	24000000
+			bclk_freq	4800000
+			tdm_slot_width	32
+		}
+
+		# include DAI copier components
+		Object.Widget.copier."0" {
+			index 1
+			dai_index 1
+			type "dai_in"
+			dai_type "SSP"
+			copier_type "SSP"
+			direction "playback"
+			stream_name "SSP0-Codec"
+			period_sink_count 0
+			period_source_count 2
+			format s16le
+		}
+
+		Object.Widget.copier."1" {
+			index 2
+			dai_index 2
+			type "dai_out"
+			dai_type "SSP"
+			copier_type "SSP"
+			direction "capture"
+			stream_name "SSP0-Codec"
+			period_sink_count 2
+			period_source_count 0
+			format s16le
+		}
+	}
+	SSP."1" {
+		id 		1
+		direction	"playback"
+		name		SSP1-Codec
+		default_hw_conf_id	0
+		sample_bits		16
+
+		Object.Base.hw_config."SSP1" {
+			id	0
+			mclk_freq	24000000
+			bclk_freq	4800000
+			tdm_slot_width	32
```

Comment:
same here, when you have a 24 MHz clock you have to use multiple of 25.

Reply:
updated the changes - referring to
https://github.com/thesofproject/sof/blob/main/tools/topology/topology1/sof-jsl-rt5682.m4#L118-L121
https://github.com/thesofproject/sof/blob/main/tools/topology/topology1/platform/intel/jsl-rt1015.m4#L11-L18